examples/custom_ct2_model.py

```diff
@@ -0,0 +1,170 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+from dataclasses import dataclass
+from functools import lru_cache
+from crossfit import op
+from crossfit.backend.torch.hf.model import HFModel
+import dask_cudf
+import crossfit as cf
+
+import ctranslate2
+from transformers import AutoConfig, AutoTokenizer
+
+
+@dataclass
+class TranslationConfig:
+    pretrained_model_name_or_path: str
+    ct2_model_path: str
+
+
+class CT2CustomModel:
+    def __init__(self, config: TranslationConfig, device="cuda"):
+        self.config = config
+        self.tokenizer = AutoTokenizer.from_pretrained(
+            pretrained_model_name_or_path=config.pretrained_model_name_or_path,
+            trust_remote_code=True,
+        )
+        self.model = ctranslate2.Translator(model_path=config.ct2_model_path, device=device)
+
+    def clean_extra_tokens(self, token_2d):
+        results = []
+        for token_1d in token_2d:
+            result = []
+            for t in token_1d:
+                if (
+                    t == self.tokenizer.pad_token
+                    or t == self.tokenizer.bos_token
+                    or t == self.tokenizer.eos_token
+                    or t == self.tokenizer.unk_token
+                ):
+                    pass
+                else:
+                    result.append(t)
+            results.append(result)
+        return results
+
+    def __call__(self, batch):
+        token_ids_2d = batch["input_ids"]
+        token_ids_1d = token_ids_2d.view(-1).tolist()
+        tokens_1d = self.tokenizer.convert_ids_to_tokens(token_ids_1d)
+        tokens_2d = [
+            tokens_1d[i : i + token_ids_2d.size(1)]
+            for i in range(0, len(tokens_1d), token_ids_2d.size(1))
+        ]
+        tokenss = self.clean_extra_tokens(tokens_2d)
+
+        tr_res = self.model.translate_batch(
+            tokenss,
+            min_decoding_length=0,
+            max_decoding_length=256,
+            beam_size=5,
+            num_hypotheses=1,
+        )
+        translations = ["".join(x.hypotheses[0]) for x in tr_res]
+        return translations
+
+
+class ModelForSeq2SeqModel(HFModel):
+    def __init__(self, config):
+        self.trans_config = config
+        self.config = AutoConfig.from_pretrained(
+            pretrained_model_name_or_path=self.trans_config.pretrained_model_name_or_path,
+            trust_remote_code=True,
+        )
+        super().__init__(
+            self.trans_config.pretrained_model_name_or_path, model_output_type="string"
+        )
+
+    def load_model(self, device="cuda"):
+        model = CT2CustomModel(self.trans_config)
+        return model
+
+    def load_config(self):
+        return AutoConfig.from_pretrained(
+            pretrained_model_name_or_path=self.trans_config.pretrained_model_name_or_path,
+            trust_remote_code=True,
+        )
```

Comment:
Use the load_cfg function, dont repeat code.

Reply:
Using load_cfg() now